### ISSUE TITLE
dump-svg always keep the original color format

### ISSUE THREAD

**Issue by yisibl** [RazrFalcon/resvg#460]
Currently, after outputting SVG with --dump-svg, the color will always be converted to rgba, which actually increases the size of the SVG file. After resvg processing, we have to perform SVG compression again.

input:

```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100">
  <rect fill="yellow" x="40" y="10" width="20" height="20"/>
  <rect fill="red" x="40" y="40" width="20" height="20" />
  <rect fill="blue" x="40" y="70" width="20" height="20" />
</svg>
```

--dump-svg output:

```svg
<svg width="100" height="100" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
    <defs/>
    <path fill="rgba(255,255,0,255)" stroke="none" d="M 40 10 L 60 10 L 60 30 L 40 30 Z"/>
    <path fill="rgba(255,0,0,255)" stroke="none" d="M 40 40 L 60 40 L 60 60 L 40 60 Z"/>
    <path fill="rgba(0,0,255,255)" stroke="none" d="M 40 70 L 60 70 L 60 90 L 40 90 Z"/>
</svg>
```

I also hope that the extra `stroke="none"` can be removed.

**Comment by RazrFalcon** [comment]
usvg output is designed to be simple, not small. Sorry.

**Comment by yisibl** [comment]
Sorry, I don’t quite understand, will always add stroke="none" to make rendering simple?

**Comment by RazrFalcon** [comment]
Not rendering, but parsing. Stroke must be set, because otherwise a file like this would be broken:

```xml
<g stroke="red">
    <path/> <!--must be none, not red -->
</g>
```

usvg doesn't care about output file size by design. It cannot be fixed.

**Comment by JoKalliauer** [comment]
> ... which actually increases the size of the SVG file.
> 
> I also hope that the extra `stroke="none"` can be removed.

For file-size-reduction use https://github.com/RazrFalcon/svgcleaner (or [alternatives](https://commons.wikimedia.org/wiki/Help:SVG_guidelines#Using_Optimizers)) after (or instead) of usvg


```svgcleaner Input.svg Output.svg  --indent 1```
or more sophisticated options such as:
```bash
svgcleaner Input.svg Output.svg --indent 1 --resolve-use no --apply-transform-to-gradients yes --apply-transform-to-shapes yes --convert-shapes yes --group-by-style no --join-arcto-flags no --join-style-attributes no --merge-gradients yes --regroup-gradient-stops yes --remove-comments yes --remove-declarations no --remove-default-attributes yes --remove-desc yes --remove-dupl-cmd-in-paths yes --remove-dupl-fegaussianblur yes --remove-dupl-lineargradient yes --remove-dupl-radialgradient yes --remove-gradient-attributes yes --remove-invalid-stops yes --remove-invisible-elements no --remove-metadata yes --remove-needless-attributes yes --remove-nonsvg-attributes no --remove-nonsvg-elements no --remove-text-attributes no --remove-title yes --remove-unreferenced-ids yes --remove-unresolved-classes yes --remove-unused-coordinates yes --remove-unused-defs yes --remove-version yes --remove-xmlns-xlink-attribute yes --simplify-transforms yes --trim-colors yes --trim-ids yes --trim-paths yes --ungroup-defs yes --ungroup-groups yes --use-implicit-cmds yes --list-separator comma --paths-to-relative yes --remove-unused-segments yes --convert-segments yes --apply-transform-to-paths yes --coordinates-precision 2 --paths-coordinates-precision 5 --properties-precision 3 --transforms-precision 7
```

This results in:
```xml
<svg height="100" viewBox="0 0 100 100" width="100" xmlns="http://www.w3.org/2000/svg">
 <path d="m40 10h20v20h-20z" fill="#ff0"/>
 <path d="m40 40h20v20h-20z" fill="#f00"/>
 <path d="m40 70h20v20h-20z" fill="#00f"/>
</svg>
```

**Comment by yisibl** [comment]
Thanks, it looks like I can only do SVG compression once more.

**Comment by JoKalliauer** [comment]
> Thanks, it looks like I can only do SVG compression once more.

If you want to compress SVG, `usvg` is the wrong program.
usvg is (in my words) a program to avoid rendering-issues, see https://github.com/RazrFalcon/resvg/tree/master/usvg#usvg for more details.
I personally use `usvg` for provide workarounds for [librsvg-bugs](https://commons.wikimedia.org/wiki/Librsvg_bugs) for Wikimedia-files (which are used in Wikipedia). [It is imho not what usvg is aiming for since the svg-files will be stored, therefore I do some post-processing.]

Quoting https://github.com/RazrFalcon/resvg/tree/master/usvg#how-to-ensure-that-svg-is-a-valid-micro-svg :
> The idea is that you should not store files produced by _usvg_. You should use them immediately.

So in other words `usvg` is to provide a dump-code that can be rendered fast and "without" rendering-bugs, but it is imho not really intended to edit or look at the dump-code. (maybe except for bug-tracing)
`usvg` will lead in maybe in 95% of all cases to a increase of the file-size, especially if you had editable font-text, the file-size can be several times larger than before.

**Comment by yisibl** [comment]
@JoKalliauer 

My function is to convert SVG so that it can be used in OpenType fonts. On the basis of usvg, I still have more things to do, compression is just one small step. The ultimate goal is to use rust to accomplish what [picosvg](https://github.com/googlefonts/picosvg) does.
